client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteSchemasAdmin.java

```diff
@@ -0,0 +1,218 @@
+
+    /**
+     * Updates an existing schema.
+     *
+     * @param schema the schema to update
+     * @return result of the operation
+     */
+    SchemaOpResult update(Schema schema);
+
+    /**
+     * Updates an existing schema, if it does not exist.
+     * If force is false, bypasses version checking.
+     *
+     * @param schema the schema to update
+     * @return result of the operation
+     */
+    SchemaOpResult update(Schema schema, boolean force);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#update(Schema)}
+     */
+    CompletionStage<SchemaOpResult> updateAsync(Schema schema);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#update(Schema, boolean)}
+     */
+    CompletionStage<SchemaOpResult> updateAsync(Schema schema, boolean force);
+
+    /**
+     * Creates or updates a schema.
+     *
+     * @param schema the schema to create or update
+     * @return result of the operation
+     */
+    SchemaOpResult createOrUpdate(Schema schema);
+
+    /**
+     * Creates or updates a schema, regardless of the version of the schema.
+     *
+     * @param schema the schema to create or update
+     * @return result of the operation
+     */
+    SchemaOpResult createOrUpdate(Schema schema, boolean force);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#createOrUpdate(Schema)}
+     */
+    CompletionStage<SchemaOpResult> createOrUpdateAsync(Schema schema);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#createOrUpdate(Schema, boolean)}
+     */
+    CompletionStage<SchemaOpResult> createOrUpdateAsync(Schema schema, boolean force);
+
+    /**
+     * Creates or updates schemas from descriptor source.
+     *
+     * @param fileDescriptorSource source of schema descriptors
+     * @return map of schema names to results
+     */
+    Map<String, SchemaOpResult> createOrUpdate(FileDescriptorSource fileDescriptorSource);
+
+    /**
+     * Deletes a schema.
+     *
+     * @param schemaName the name of the schema
+     * @return result of the operation
+     */
+    SchemaOpResult remove(String schemaName);
+
+    /**
+     * Deletes a schema, with option to force deletion.
+     *
+     * @param schemaName the name of the schema
+     * @param force whether to force delete
+     * @return result of the operation
+     */
+    SchemaOpResult remove(String schemaName, boolean force);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#remove(String)}
+     */
+    CompletionStage<SchemaOpResult> removeAsync(String schemaName);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#remove(String, boolean)}
+     */
+    CompletionStage<SchemaOpResult> removeAsync(String schemaName, boolean force);
+
+    /**
+     * Checks if a schema exists.
+     *
+     * @param schemaName the name of the schema
+     * @return true if exists, false otherwise
+     */
+    boolean exists(String schemaName);
```

Comment:
I don't see the need for this method as the user can invoke `get()`

Reply:
it allows providing a way to check if exists without bringing the schema to the client.